### PR TITLE
refactor(utils.lua): update deprecated api

### PR DESCRIPTION
**Pull request by horaoen**
update deprecated api for "client.resolved_capabilities" and "vim.lsp.buf.formatting_sync"

### CODE REVIEW

File: lua/nvim-lsp-setup/utils.lua [hunk 1]
```diff
@@ -31,16 +31,16 @@ function M.default_mappings(bufnr, mappings)
...
 
 function M.disable_formatting(client)
-    client.resolved_capabilities.document_formatting = false
-    client.resolved_capabilities.document_range_formatting = false
+    client.server.document_formatting = false
+    client.server.document_range_formatting = false
 end
 
 function M.format_on_save(client)
     if client.resolved_capabilities.document_formatting then
```
Comment (mcginleyr1) [inline lua/nvim-lsp-setup/utils.lua:39]: Should this be changed too?

File: lua/nvim-lsp-setup/utils.lua [hunk 2]
```diff
@@ -31,16 +31,16 @@ function M.default_mappings(bufnr, mappings)
 end
 
 function M.disable_formatting(client)
-    client.resolved_capabilities.document_formatting = false
-    client.resolved_capabilities.document_range_formatting = false
+    client.server.document_formatting = false
```
Comment (mcginleyr1) [inline lua/nvim-lsp-setup/utils.lua:34]: shoudl this be server_capabilities ?

Comment (horaoen) [reply lua/nvim-lsp-setup/utils.lua]: sorry, I have resolved this spelling errors.

File: lua/nvim-lsp-setup/utils.lua [hunk 3]
```diff
@@ -31,16 +31,16 @@ function M.default_mappings(bufnr, mappings)
...
+    client.server_capabilities.document_range_formatting = false
 end
 
 function M.format_on_save(client)
-    if client.resolved_capabilities.document_formatting then
+    if client.server_capabilities.document_formatting then
         vim.cmd([[
           augroup Format
             au! * <buffer>
-            au BufWritePre <buffer> lua vim.lsp.buf.formatting_sync(nil, 1000)
```
Comment (junnplus) [inline lua/nvim-lsp-setup/utils.lua:43]: Seems not compatible with nvim 0.7 ?

Comment (horaoen) [reply lua/nvim-lsp-setup/utils.lua:43]: ,my neovim‘s version is 0.8, it works seems well.

Comment (junnplus) [reply lua/nvim-lsp-setup/utils.lua:43]:
```
E5108: Error executing lua [string ":lua"]:1: attempt to call field 'format' 
```

We should be compatible with neovim 0.7

Comment (horaoen) [reply lua/nvim-lsp-setup/utils.lua]: thank you, I have changed it back.

**Pull request closed without merging**